packages/canvas-runtime-utils/src/utils.ts

```diff
@@ -162,70 +162,52 @@ export function getStylesAlias(id: string, componentTree: Tree, cssTree: Tree) {
+
+  // Base case: if the selector array is empty, set the value of the new object to the given value
+  if (selector.length === 0) {
     return value;
   }
-  if (keys.length === 1) {
-    referenceObject[keys[0]] = value;
-  } else {
-    const [key, ...remainingKeys] = keys;
-    const nextKey = remainingKeys[0];
-    const nextRemainingKeys = remainingKeys.slice(1);
-
-    if (
-      referenceObject[key] !== undefined &&
-      typeof nextKey === "string" &&
-      Array.isArray(referenceObject[key])
-    ) {
-      throw "Datatype of selector and reference object does not match";
-    }
 
-    if (
-      referenceObject[key] !== undefined &&
-      typeof nextKey === "number" &&
-      typeof referenceObject[key] === "object"
-    ) {
-      throw "Datatype of selector and reference object does not match";
+  // Recursive case: update the value of the property or element specified by the first element of the selector array
+  const [key, ...remainingKeys] = selector;
+  if (typeof key === "string") {
+    if (typeof newObj === "string") return value;
+    // If the key is a string, it refers to an object property
+    if (typeof newObj !== "object") {
+      throw new Error("Cannot access property of non-object or string");
     }
-
-    if (typeof nextKey === "number") {
-      if (referenceObject[key] === undefined || referenceObject[key] === null) {
-        // create array
-        referenceObject[key] = [];
-      }
-
-      // Fill empty index with empty object
-      if (nextKey <= referenceObject[key].length) {
-        const delta = nextKey + 1 - referenceObject[key].length;
-        for (let i = 0; i < delta; i++) {
-          referenceObject[key].push({});
-        }
-      } else {
-        throw "Cannot add index that is greater than the array size";
-      }
-
-      // recursively write the object
-      referenceObject[key][nextKey] = createObject(
-        referenceObject[key][nextKey],
-        nextRemainingKeys,
-        value
-      );
+    if (!newObj.hasOwnProperty(key)) {
+      newObj[key] = {};
+    }
+    newObj[key] = createObject(newObj[key], remainingKeys, value);
+  } else if (typeof key === "number") {
+    // If the key is a number, it refers to an array element
+    if (!Array.isArray(newObj)) {
+      throw new Error("Cannot access element of non-array");
+    }
+    // If the key is 0 (first element) or equal to the length of the array (last element), set the value at the specified location in the array
+    if ((key === 0 || key === newObj.length) && selector.length == 1) {
+      newObj[key] = value;
     } else {
-      // recursively write the object
-      referenceObject[key] = createObject(
-        typeof referenceObject[key] === "undefined" ? {} : referenceObject[key],
-        remainingKeys,
-        value
-      );
+      if (key > newObj.length) {
+        throw new Error("Array index out of bounds");
+      }
+      newObj[key] = createObject(newObj[key], remainingKeys, value);
```

Comment:
This line will replace the old value with the new value if the index is already present

Reply:
By reading the code, it seems like this won't be able to handle the case where we are changing an entry in between the array. It will throw an error because the recursive call to `createObject()` in this line will be passed an empty `remainingKeys` array. This will lead to an undefined `key` in this call.

Reply:
If the key is undefined it will return the value so it finally gets patched

Reply:
Are you sure we won't hit the else condition in line 208?

Reply:
We will hit the condition in line 208 only if we pass an object in the selector array, or a nested array basically anything other than a string or a number. As it checks on line 178 if the selector array is empty, if it is it will return the value and will not proceed to the line where it extracts the key from the selector array